src/app/components/Dialogs/TxDialog.tsx

```diff
@@ -29,46 +29,46 @@ interface Props {
 }
 
 export function TxDialog(props: Props) {
+  const { tx, onUserConfirmed, onSuccess } = props;
+
   const { t } = useTranslation();
   const { address } = useWalletContext();
   const close = () => {
```

Comment:
Make this and `confirm` one-liners (remove unnecessary parentheses) and also either move it under the last hook or use `useCallback`.